libphobos/libdruntime/gcc/sections/windows.d

```diff
@@ -167,33 +129,32 @@ void finiTLSRanges(void[] rng) nothrow @nogc
 
 void scanTLSRanges(void[] rng, scope void delegate(void* pbeg, void* pend) nothrow dg) nothrow
 {
-    if (conservative)
+    version (GNU_EMUTLS)
     {
-        dg(rng.ptr, rng.ptr + rng.length);
+        import gcc.emutls;
+        _d_emutls_scan(dg);
     }
     else
     {
-        for (auto p = &_TP_beg; p < &_TP_end; )
-        {
-            uint beg = *p++;
-            uint end = beg + cast(uint)((void*).sizeof);
-            while (p < &_TP_end && *p == end)
-            {
-                end += (void*).sizeof;
-                p++;
-            }
-            dg(rng.ptr + beg, rng.ptr + end);
-        }
+        static assert(false, "Architecture not supported.");
     }
 }
 
+/*
+ * Currently unused. As this is called by the compiler, we
+ * still have to provide a dummy function here.
+ */
+extern(C) void _d_dso_registry(void* data)
+{
```

Comment:
Maybe rather than doing nothing, move the the entire of `getModuleInfos()` here, using `data._minfo_beg .. data._minfo_end` as the start/stop brackets instead of the globals, and finishing off with `_sections._moduleGroup = ModuleGroup(result)`

Reply:
We could do that, though I wanted to keep the code close to upstream for now, at least as long as we're not using the `_d_dso_registry` for shared libraries anyway (proper support for shared MinGW libraries would be much more work, I guess).